go-controller/pkg/kubevirt/dhcp_test.go

```diff
@@ -0,0 +1,148 @@
+package kubevirt
+
+import (
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+var _ = Describe("Kubevirt", func() {
+	type dhcpTest struct {
+		cidrs                             []string
+		controllerName, namespace, vmName string
+		dns                               *corev1.Service
+		expectedDHCPConfigs               dhcpConfigs
+		expectedError                     string
+	}
+	var (
+		svc = func(namespace string, name string, clusterIPs []string) *corev1.Service {
+			return &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "kube-system",
+					Name:      "kube-dns",
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIPs: clusterIPs,
+				},
+			}
+		}
+		key = func(namespace, name string) ktypes.NamespacedName {
+			return ktypes.NamespacedName{Namespace: namespace, Name: name}
+		}
+		parseCIDR = func(cidr string) *net.IPNet {
+			_, parsedCIDR, err := net.ParseCIDR(cidr)
+			Expect(err).ToNot(HaveOccurred())
+			return parsedCIDR
+		}
+	)
+	DescribeTable("composing dhcp options should success", func(t dhcpTest) {
+		svcs := []corev1.Service{}
+		if t.dns != nil {
+			svcs = append(svcs, *t.dns)
+		}
+		fakeClient := &util.OVNMasterClientset{
+			KubeClient: fake.NewSimpleClientset(&corev1.ServiceList{
+				Items: svcs,
+			}),
+		}
+		watcher, err := factory.NewMasterWatchFactory(fakeClient)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(watcher.Start()).To(Succeed())
+
+		cidrs := []*net.IPNet{}
+		for _, cidr := range t.cidrs {
+			cidrs = append(cidrs, parseCIDR(cidr))
+		}
+		obtaineddhcpConfigs, err := composeDHCPConfigs(watcher, t.controllerName, ktypes.NamespacedName{Namespace: t.namespace, Name: t.vmName}, cidrs)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(t.expectedDHCPConfigs.V4).To(Equal(obtaineddhcpConfigs.V4))
+		Expect(t.expectedDHCPConfigs.V6).To(Equal(obtaineddhcpConfigs.V6))
```

Comment:
nit: these expectations are "upside down"; the `Expect` argument is the actual. You're meant to compare "actual" against expected, right ?